src/kernel.js

```diff
@@ -4,26 +4,36 @@ const Matrix = require('ml-matrix');
 
 const GaussianKernel = require('ml-gaussian-kernel');
 const PolynomialKernel = require('ml-polynomial-kernel');
+const ANOVAKernel = require('./kernels/anova-kernel');
+const CauchyKernel = require('./kernels/cauchy-kernel');
+const ExponentialKernel = require('./kernels/exponential-kernel');
+const HistogramKernel = require('./kernels/histogram-intersection-kernel');
+const LaplacianKernel = require('./kernels/laplacian-kernel');
+const MultiquadraticKernel = require('./kernels/multiquadratic-kernel');
+const RationalKernel = require('./kernels/rational-quadratic-kernel');
 
 class Kernel {
     constructor(type, options) {
+        var kernelType = {  'gaussian': new GaussianKernel(options),
```

Comment:
This is not very efficient. You are creating a lot of instances just to use one of them in the end. I suggest you do it like that instead:

```js
// outside of the Kernel constructor
const kernelTypes = {
    gaussian: GaussianKernel,
    rbf: GaussianKernel,
    polynomial: PolynomialKernel,
    // ...
};

class Kernel {
    constructor(type, options) {
        if (typeof type === 'string') {
            type = type.toLowerCase();
            var KernelConstructor = kernelTypes[type];
            if (KernelConstructor) {
                this.kernelFunction = new KernelConstructor(options);
            } else {
                throw new Error('unsupported kernel type: ' + type);
            }
        } else if ...
    }
}
```